contracts/root/TokenPredicates/ERC1155Predicate.sol

```diff
@@ -84,20 +111,72 @@ contract ERC1155Predicate is ITokenPredicate, ERC1155Receiver, AccessControlMixi
             uint256[] memory amounts,
             bytes memory data
         ) = abi.decode(depositData, (uint256[], uint256[], bytes));
+
+        IERC1155 token = IERC1155(rootToken);
+
+        address[] memory addrArray = makeArrayWithAddress(address(this), ids.length);
+        uint256[] memory oldBalances = token.balanceOfBatch(addrArray, ids);
+        token.safeBatchTransferFrom(
+            depositor,
+            address(this),
+            ids,
+            amounts,
+            data
+        );
+        uint256[] memory lockedBalances = calculateLockedAmounts(
+            oldBalances, 
+            token.balanceOfBatch(addrArray, ids));
+
         emit LockedBatchERC1155(
             depositor,
             depositReceiver,
             rootToken,
             ids,
-            amounts
+            lockedBalances
         );
-        IERC1155(rootToken).safeBatchTransferFrom(
+    }
+
+    function verifiedLockTokens(
+        address depositor,
+        address depositReceiver,
+        address rootToken,
+        bytes calldata depositData
+    )
+        external
+        override
+        only(MANAGER_ROLE)
+        returns(bytes memory)
+    {
+        // forcing batch deposit since supporting both single and batch deposit introduces too much complexity
+        (
+            uint256[] memory ids,
+            uint256[] memory amounts,
+            bytes memory data
+        ) = abi.decode(depositData, (uint256[], uint256[], bytes));
+
+        IERC1155 token = IERC1155(rootToken);
+
+        address[] memory addrArray = makeArrayWithAddress(address(this), ids.length);
+        uint256[] memory oldBalances = token.balanceOfBatch(addrArray, ids);
+        token.safeBatchTransferFrom(
             depositor,
             address(this),
             ids,
             amounts,
             data
         );
+        uint256[] memory lockedBalances = calculateLockedAmounts(
+            oldBalances, 
+            token.balanceOfBatch(addrArray, ids));
+
+        emit LockedBatchERC1155(
+            depositor,
+            depositReceiver,
+            rootToken,
+            ids,
+            lockedBalances
+        );
+        return abi.encode(ids, lockedBalances, data);
```

Comment:
Instead of copy-pasting the same code, user `verifiedLockTokens` inside `lockTokens`, do the same for other predicates too, please

Reply:
Yes, did this in latest commit.